service/cloudconfig/service.go

```diff
@@ -104,5 +106,5 @@ func newCloudConfig(template string, params k8scloudconfig.Params) (string, erro
 		return "", microerror.Mask(err)
 	}
 
-	return cloudConfig.Base64(), nil
+	return base64.StdEncoding.EncodeToString([]byte(cloudConfig.String())), nil
```

Comment:
The .Base64() method returns a gzip compressed cloud-config, but Azure does not seem to allow that.

I has to be an uncompressed base64 encoded cloud-config :(

Reply:
It's a shame gzip compression isn't supported as I know we're close to the 64kb limit. Unless we can reduce it we may need to download the cloudconfig from blob storage but for now lets go with this.

Reply:
Something we could do if we want to avoid downloading it from blob storage would be to compress the file contents, like here: https://coreos.com/os/docs/latest/cloud-config.html#write_files.

If we decide to do so, this PR would still be working properly :)